beacon_node/execution_layer/src/lib.rs

```diff
@@ -374,7 +374,8 @@ impl<T: EthSpec> ExecutionLayer<T> {
 
     /// Attempt to retrieve a full payload from the payload cache by the payload root
     pub fn get_payload_by_root(&self, root: &Hash256) -> Option<ExecutionPayload<T>> {
-        self.inner.payload_cache.pop(root)
+        let payload_cache = self.inner.payload_cache.clone();
```

Comment:
this clones the whole cache but I think it would be better to clone the entry, by adding a new method to `PayloadCache` which uses `get` and then `clone`